src/core_modules/capture-core/components/WidgetsChangelog/common/ChangelogFilterBar/DropdownFilter.js

```diff
@@ -0,0 +1,79 @@
+// @flow
+import React, { useCallback } from 'react';
+import { DropdownButton, FlyoutMenu, MenuItem, Divider, colors } from '@dhis2/ui';
+import { withStyles } from '@material-ui/core';
+import i18n from '@dhis2/d2-i18n';
+import type { FilterValueType } from './ChangelogFilter.types';
+
+type FilterItem = {
+    id: string,
+    name: string,
+};
+
+type Props = {
+    label: string,
+    items: Array<FilterItem>,
+    classes: Object,
+    filterColumn: string,
+    openMenuName: string | null,
+    onToggleMenu: (menuName: string) => void,
+    onItemSelected: (value: FilterValueType, filterColumn: string) => void,
+    selectedFilterValue: FilterValueType,
+};
+
+const styles = {
+    label: {
+        color: colors.grey600,
+    },
+};
+
+const DropdownFilterPlain = ({
+    label,
+    classes,
+    items,
+    filterColumn,
+    openMenuName,
+    onToggleMenu,
+    onItemSelected,
+    selectedFilterValue,
+}: Props) => {
+    const isMenuOpen = openMenuName === filterColumn;
```

Comment:
The menu does not close when filtering for the same value the second time. Could you have a look? Thanks!

https://github.com/user-attachments/assets/94645d70-a8be-4c63-94f8-31d84e07f93e